opus/application/static_media/js/browse.js

```diff
@@ -424,6 +462,59 @@ var o_browse = {
         });
     }, // end browse behaviors
 
+    // update order arrows right away when user clicks on sorting arrows in pill or table header
+    // sync up arrows in both sorting pill and table header
+    updateOrderIndicator: function(headerOrderIndicator, pillOrderIndicator, isDescending, slug) {
+        let headerOrder = isDescending ? "sort-desc" : "sort-asc";
+        let headerOrderArrow = isDescending ? tableSortUpArrow : tableSortDownArrow;
+        let isPillOrderDesc = isDescending ? "true" : "false";
+        let pillOrderTooltip = isDescending ? "Change to ascending sort" : "Change to descending sort";
+        let pillOrderArrow = isDescending ? pillSortUpArrow : pillSortDownArrow;
+
+        // If header already exists, we update the header arrow, else we do nothing
+        if (headerOrderIndicator && headerOrderIndicator.length !== 0) {
+            headerOrderIndicator.data("sort", `${headerOrder}`);
+            headerOrderIndicator.attr("class", `column_ordering ${headerOrderArrow}`);
+
+            // Reset arrows on rest of table headers
+            // let headers = $(`.dataTable th a:not([data-slug="opusid"], [data-slug=${slug}])`).find("span:last");
+            let headers = $(`.dataTable th a:not([data-slug=${slug}])`).find("span:last");
+            headers.data("sort", "none");
+            headers.attr("class", defaultTableSortArrow);
+        }
+
+        // If pill already exists, we update the pill, else we re-render the pill
+        if (pillOrderIndicator.length !== 0 && slug !== "opusid") {
```

Comment:
Why are you treating opusid specially here? Why don't we just flip the arrow on the opusid pill like we do any other?

Reply:
This causes a really weird bug as written. Add Right Ascension (Min) and Right Ascension (Max) as columns. Go to the table and sort on Right Ascension (Min). Then go to select metadata and remove Right Ascension (Min). Now click on the opus id pill to swap the sort order. Due to this line, it will need to rerender all the pills. But below it gets the name of the pill from the table header - but the column isn't in the table anymore! It will thus do something weird until the data returns, at which point we once again redraw all the pills due to updateSortOrder() and that will get the names right.

Reply:
The reason of treating opusid specially is for the case when we click "OPUS ID" table header and there are two sorting pills displayed. At this time, we only want to display "OPUS ID" pill, so we have to re-render the pill again (to remove the extra one pill). 

For the bug mentioned above, it was caused by method I used to get the label. It was creating the label by chopping all the string after "(" to remove the unit, but "(Max)" and "(Min)" also got chopped out. 

I follow Debby's feedback to store the proper label in data-label for each table header, and modify the logic of getting labels. It will now only remove "(" with units and keep "(Max)" and "(Min)" in label. (line 1194-1198 in browse.js)

Reply:
As we discussed in person, the easier way to handle this is just to re-render all the pills each time. That will make the code a lot less complex and also get rid of the dependence on opusid as a special case. Let's go ahead and do that now so the code is cleaner.